src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,256 @@
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+            expect(zipClient.Checkout.init).toHaveBeenCalledWith(zipInitPayload);
+        });
+
+        it('fails to execute the strategy if the zip client was not loaded', async () => {
+            jest.spyOn(zipScriptLoader, 'load')
+                .mockReturnValueOnce(Promise.resolve(undefined));
+            await strategy.initialize(zipOptions);
+            try {
+                await strategy.execute(orderRequestBody, zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(NotInitializedError);
+                expect(orderActionCreator.submitOrder).not.toHaveBeenCalled();
+                expect(paymentActionCreator.submitPayment).not.toHaveBeenCalled();
+                expect(store.dispatch).not.toHaveBeenCalledWith(submitOrderAction);
+                expect(store.dispatch).not.toHaveBeenCalledWith(submitPaymentAction);
+                expect(zipClient.Checkout.init).not.toHaveBeenCalled();
+            }
+        });
+
+        it('fails to execute the strategy if lightbox fails to provide a checkoutId', async () => {
+            const noIdZipClient = getZipScriptMock('noCheckoutId');
+            jest.spyOn(zipScriptLoader, 'load')
+                .mockReturnValueOnce(Promise.resolve(noIdZipClient));
+            await strategy.initialize(zipOptions);
+            try {
+                await strategy.execute(orderRequestBody, zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(PaymentMethodInvalidError);
+                expect(paymentActionCreator.submitPayment).not.toHaveBeenCalled();
+                expect(store.dispatch).not.toHaveBeenCalledWith(submitPaymentAction);
+            }
+        });
+
+        it('cancels the strategy execution if the lightbox is closed', async () => {
+            const cancelledZipClient = getZipScriptMock('cancelled');
+            jest.spyOn(zipScriptLoader, 'load')
+                .mockReturnValueOnce(Promise.resolve(cancelledZipClient));
+            await strategy.initialize(zipOptions);
+            try {
+                await strategy.execute(orderRequestBody, zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(PaymentMethodCancelledError);
+                expect(paymentActionCreator.submitPayment).not.toHaveBeenCalled();
+                expect(store.dispatch).not.toHaveBeenCalledWith(submitPaymentAction);
+            }
+        });
+
+        it('fails to execute the strategy if lightbox response state is not handled by the strategy', async () => {
+            const erroredZipClient = getZipScriptMock('error');
+            jest.spyOn(zipScriptLoader, 'load')
+                .mockReturnValueOnce(Promise.resolve(erroredZipClient));
```

Comment:
It does not fail to load the client, it resolves a regular zip client but the lightbox response on `onComplete` response  is `state: 'error'` instead of the regular `approved` or `cancelled`.
It should be more clear after I reword this test's name.